sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/LiteralGenerator.scala

```diff
@@ -102,8 +102,14 @@ object LiteralGenerator {
   lazy val dateLiteralGen: Gen[Literal] =
     for { d <- Arbitrary.arbInt.arbitrary } yield Literal.create(new Date(d), DateType)
 
-  lazy val timestampLiteralGen: Gen[Literal] =
-    for { t <- Arbitrary.arbLong.arbitrary } yield Literal.create(new Timestamp(t), TimestampType)
+  private def yearToMillis(year: Long): Long = {
+    year * 365 * 24 * 3600 * 1000L
+  }
+
+  lazy val timestampLiteralGen: Gen[Literal] = {
+    for { millis <- Gen.choose(yearToMillis(-9999), yearToMillis(10000)) }
+      yield Literal.create(new Timestamp(millis), TimestampType)
+  }
```

Comment:
@MaxGekk . In general, we don't change the test cases during refactoring in order to ensure the test coverage. I'm wondering if this is a required change due to the known limitation of the new functions? I'd make this as an another PR.

Reply:
I changed the random generator for timestamps, and limit max/min values for milliseconds since epoch because the long random generator can produce milliseconds that caused Long overflow at conversion of milliseconds to microseconds. Internally as you know we store microseconds since epoch in `TimestampType`. For example, the old (current) generator can create an instance of `java.sql.Timestamp(-3948373668011580000, 570000000)`. New function `fromJavaTimestamp` calls `instantToMicros`, and inside of it we use `multiplyExact` which can detect `Long` overflow on multiplication:
```scala
def instantToMicros(instant: Instant): Long = {
  val us = Math.multiplyExact(-3948373668011580, 1000000)
    ...
}
```
<img width="873" alt="Screen Shot 2019-04-07 at 08 41 30" src="https://user-images.githubusercontent.com/1580697/55679740-fa961780-5910-11e9-914c-11a1336894e8.png">

Previous (current) implementation doesn't detect the overflow at all.

Reply:
The long overflow exception caused failures in https://github.com/apache/spark/blob/1d20d13149140f53df307f47420740f45b4fa5f6/sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/DateExpressionsSuite.scala#L202

Reply:
That's the exact reason you should not put that in this PR, @MaxGekk .
Please do the improvement in another PR first. And do the refactoring PR later.

Reply:
@dongjoon-hyun ok. I will open another PR. Thanks.